swift/apple/FirezoneKit/Tests/FirezoneKitTests/NetworkSettingsTests.swift

```diff
@@ -0,0 +1,328 @@
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: nil,
+      routes4: [route1],
+      routes6: []
+    )
+
+    let route2 = NEIPv4Route(destinationAddress: "192.168.2.0", subnetMask: "255.255.255.0")
+
+    // Second update with additional route
+    let result = settings.updateTunInterface(
+      ipv4: "10.0.0.1",
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: nil,
+      routes4: [route1, route2],
+      routes6: []
+    )
+
+    #expect(result?.ipv4Settings?.includedRoutes?.count == 2)
+  }
+
+  // MARK: - DNS Resources Updates After TUN Config
+
+  @Test("Same DNS resource list should not emit settings")
+  func sameDnsResourcesNoUpdate() async throws {
+    var settings = NetworkSettings()
+
+    // Configure TUN first
+    _ = settings.updateTunInterface(
+      ipv4: "10.0.0.1",
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: nil,
+      routes4: [],
+      routes6: []
+    )
+
+    // First DNS resources update (empty list)
+    _ = settings.updateDnsResources(addresses: [])
+
+    // Second DNS resources update with same empty list
+    let result = settings.updateDnsResources(addresses: [])
+
+    #expect(result == nil, "Identical DNS resource list should not emit settings")
+  }
+
+  // MARK: - Dummy Match Domain Tests
+
+  @Test("Set dummy match domain")
+  func setDummyMatchDomain() async throws {
+    var settings = NetworkSettings()
+
+    // Configure TUN first
+    _ = settings.updateTunInterface(
+      ipv4: "10.0.0.1",
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: "example.com",
+      routes4: [],
+      routes6: []
+    )
+
+    // Set dummy match domain
+    let result = settings.setDummyMatchDomain()
+
+    #expect(result?.dnsSettings?.matchDomains == ["firezone-fd0020211111"])
+    #expect(result?.dnsSettings?.searchDomains == ["example.com"])
+  }
+
+  @Test("Clear dummy match domain restores original")
+  func clearDummyMatchDomain() async throws {
+    var settings = NetworkSettings()
+
+    // Configure TUN with search domain
+    _ = settings.updateTunInterface(
+      ipv4: "10.0.0.1",
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: "example.com",
+      routes4: [],
+      routes6: []
+    )
+
+    // Set dummy
+    _ = settings.setDummyMatchDomain()
+
+    // Clear dummy
+    let result = settings.clearDummyMatchDomain()
+
+    #expect(result?.dnsSettings?.matchDomains == ["", "example.com"])
+    #expect(result?.dnsSettings?.searchDomains == ["example.com"])
+  }
+
+  @Test("Clear dummy match domain without search domain")
+  func clearDummyMatchDomainNoSearchDomain() async throws {
+    var settings = NetworkSettings()
+
+    // Configure TUN without search domain
+    _ = settings.updateTunInterface(
+      ipv4: "10.0.0.1",
+      ipv6: "fd00::1",
+      dnsAddresses: ["1.1.1.1"],
+      searchDomain: nil,
+      routes4: [],
+      routes6: []
+    )
+
```

Comment:
Same here re: adding an expect to assert on the state pre-function-under-test call.

Reply:
Same as above: There is another test that asserts that behaviour. I prefer not asserting things mid-test to ensure a test only breaks if the invariant it is trying to capture is broken and not some other part of the system.